src/NuGet.Core/NuGet.Commands/RestoreCommand/OriginalCaseGlobalPackageFolder.cs

```diff
@@ -75,7 +75,7 @@ public async Task CopyPackagesToOriginalCaseAsync(IEnumerable<RestoreTargetGraph
                 foreach (var remoteMatch in packages)
                 {
                     var identity = GetPackageIdentity(remoteMatch);
-                    var hashPath = _pathResolver.GetHashPath(identity.Id, identity.Version);
+                    var hashPath = _pathResolver.GetNewHashPath(identity.Id, identity.Version);
```

Comment:
What is `NewHashPath`? Can we have this be a more descriptive name? Maybe something like `GetNupkgMetadataPath`?